File: C++ ad y ad/3_ Orientado a objetos/S04.1_ Estrellas en el cielo/Solucion_estrellas_en_el_cielo/Estrellas_en_el_cielo/estrella.cpp
#include "estrella.h"

#include <iostream>

#include "mgtv_library.h"

using namespace std;
using namespace mgtv_library::console;


int Estrella::cantCambiosPosicion = 0;

Estrella::Estrella()
{
	posicion = { 0, 0 };
}
Estrella::~Estrella()
{
	cout << "La estrella ubicada en la posicion (" << posicion.x << "; " << posicion.y << ") ha sido liberada de la memoria.\n\n";
}


int Estrella::getCantCambiosPosicion()
{
	return cantCambiosPosicion;
}

void Estrella::draw()
{
	ConsoleExt::goToCoordinates(posicion.x, posicion.y);
	ConsoleExt::writeWithColor("*", COLOR::C_YELLOW);
}

POSICION Estrella::getPosicion()
{
	return posicion;
}
void Estrella::setPosicion(int x, int y)
{
	posicion = { x, y };

	cantCambiosPosicion++;
}

File: C# ad y ad/2_ Orientado a objetos/S04.1_ Estrellas en el cielo/Solucion_estrellas_en_el_cielo/Estrellas_en_el_cielo/Estrella.cs
using System;

using Mgtv_Library.Console;


namespace Estrellas_en_el_cielo
{
	class Estrella
	{
		static int cambiosPosicion = 0;

		Posicion posicion;


		public Estrella()
		{
			posicion = new Posicion();
		}


		public static int GetCambiosDePosicion()
		{
			return cambiosPosicion;
		}

		public void Draw()
		{
			ConsoleExt.GoToCoordinates(posicion.x, posicion.y);
			ConsoleExt.WriteInColor("*", ConsoleColor.Yellow);
		}

		public Posicion GetPosicion()
		{
			return posicion;
		}
		public void SetPosicion(int x, int y)
		{
			posicion.x = x;
			posicion.y = y;

			cambiosPosicion++;
		}
	}
}
